new/kitchen_telemetry.md

```diff
@@ -0,0 +1,143 @@
+---
+RFC: unassigned
+Title: Use of telemetry in Test Kitchen
+Author: Thom May <thom@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Use of telemetry in Test Kitchen
+
+This RFC is a description of Test Kitchen's implementation of <the
+telemetry RFC, as yet unumbered>, as required in that RFC.
+
+## What questions do we hope to answer?
+
+ * What platforms do our users run kitchen on?
+ * Which plugins are used by kitchen users?
+ * Which bento boxes are used?
+ * Which provisioners are used?
+ * If a chef provisioner is used, which chef versions are installed?
+ * Which commands are used?
+
+## Implementation
+
+The bulk of this RFC is the description of the events, below. We
+envisage that a single client library will be built for all chef
+projects, and Kitchen will consume that to send events. That library
+would ensure events are sent in a non-blocking manner, and would deal
+with retries, session handling and opt-out.
```

Comment:
Yes, the client will do exponential back off, and will probably only do a relatively small number of retries. I think it's reasonable to expect that once a client detects it can't connect to the stats endpoint it'll stop trying for the rest of the lifetime of that process, but I don't think we'd disable the system from ever trying again; that's the user's responsibility.